File: src/main/java/sx/jms/Handler.java
package sx.jms;

import javax.jms.JMSException;
import javax.jms.Message;
import javax.jms.Session;

/**
 * Message handler
 * Created by masc on 28.06.15.
 */
public interface Handler<T> {
    void onMessage(T message, Message jmsMessage, Session session) throws JMSException;
}

File: src/main/java/sx/jms/Handler.kt
package sx.jms

import com.google.common.reflect.TypeToken
import java.io.Serializable
import java.lang.reflect.ParameterizedType

/**
 * Message handler
 * Created by masc on 28.06.15.
 */
interface Handler<in T> {
    /** Message types to handle. Must be overridden when multiple message types should be handled (Handler<Anx>) */
    val messageTypes: List<Class<*>> get() {
        return arrayListOf()
    }

    /** Override for serialized/object message handling */
    fun onMessage(message: T, replyChannel: Channel?)
}
